src/main/java/com/trustedchoice/askkodiak/v2/client/AskKodiak.java

```diff
@@ -105,6 +108,82 @@
     @RequestLine("GET /v2/products/class-code/naics/{code}")
     Products getProductsEligibleForCode(@Param("code") String code) throws AskKodiakException;
 
+    /**
+     * Get products eligible a given NAICS code. code can be any valid 2017 NAICS edition:
+     * <ol>
+     * <li>Sector : A NAICS Sector Code. Generally a 2 digit numeric value, but occasionally a hyphenated range for
+     * certain sectors. (e.g '11' or '44-45', representing Agriculture, Forestry, Fishing and Hunting and Retail Trade
+     * respectively). See the NAICS/GetSectors interface for a full list.</li>
+     * <li>Subsector : A 3 digit numeric NAICS Subsector code (e.g. '531', representing Real Estate)</li>
+     * <li>National Industry : A 4 digit numeric NAICS National Industry Code (e.g. '3361', representing Motor Vehicle
+     * Manufacturing)</li>
+     * <li>International Industry : A 5 digit NAICS International Industry code (e.g. '11331', representing
+     * Logging)</li>
+     * <li>National Industry : A 6 digit NAICS National Industry code (e.g. '312120', representing Breweries)</li>
+     * <li>Hash : An MD5 Hash of a six-digit National Industry code and sub-description (e.g.
+     * '03da921f2510c7ab4e47dcd1f9061264', the calculated MD5 hash of 713990Miniature golf courses). This is the most
+     * precise eligibility request that can be made. Please note, valid 2012 NAICS edition hashes will also be accepted,
+     * although they will be mapped to their 2017 equivalent. The mapped hash will be included in the response payload.
+     * The superset of pre-calculated hashes are available from the NAICS/GetCodes interface.</li>
+     * </ol>
+     * <p>
+     * The API account acts as a member of your 'staff'. Products available to this user group will be included in
+     * results.
+     * <p>
+     * In the event that the code requested is anything other than an MD5 hash of six-digit National Industry code +
+     * sub-description, products with any eligibility whatsoever in the requested NAICS group will be included in the
+     * response. Each product in the results is extended with a property representing the percentage of codes in the
+     * requested group for which the product is eligible.
+     *
+     * @param code  Any valid 2017 NAICS Code (at any level of the NAICS hierarchy) or MD5 Hash (6 Digit NAICS National
+     *              Industry Code + Sub-Description). See above for a more detailed description of possible values.
+     *              Conditional rules which pertain to the specified code will be applied to the results.
+     * @param query Query parameters
+     * @return Eligible products
+     * @throws AskKodiakException error
+     */
+    default Products getProductsEligibleForCode(@Param("code") String code, @QueryMap EligibleQuery query) throws AskKodiakException {
```

Comment:
Does it make more sense to put this logic inside a `RequestInterceptor` instead and then register that interceptor with `AskKodiakClient`? That could also fix the issue for any other `@RequestLine` methods that have query parameters containing `+`.